src/vg/civcraft/mc/namelayer/listeners/MercuryMessageListener.java

```diff
@@ -49,5 +88,29 @@ else if (reason.equals("transfer")){
 			gm.invalidateCache(g.getName());
 		}
 	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogin(PlayerJoinEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "login "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGIN");
+	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogoff(PlayerQuitEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "logoff "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGOFF");
+	}
 
+	@Override
+	public void run() {
+		NameLayerPlugin nl = NameLayerPlugin.getInstance();
+		nl.isMercuryEnabled = Bukkit.getPluginManager().isPluginEnabled("Mercury");
+		if (nl.isMercuryEnabled){
+			nl.getServer().getPluginManager().registerEvents(this, nl);
```

Comment:
add this here: getServer().getPluginManager().registerEvents(new MercuryMessageListener(), this);

Also I'll end up refactoring this so we dont need the delay but its okay for now.  I need to read up more on the documentation.